src/index.ts

```diff
@@ -199,6 +234,83 @@ export async function newEventStore<Q>(
     return _impl;
   }
 
+  function createAggregateWhile<AQ extends Q, A extends any[], T>(
+    aggregateName: string,
+    query: AQ,
+    matches: AggregateMatches<T>,
+  ): AggregateWhile<A, T> {
+
+    return async function _impl(...args: A) {
+      return async function(whileTrue: (event: any) => boolean, guid: string) {
+
+        const start = Date.now();
+
+        const id = createHash('sha256')
+          .update(JSON.stringify(query) + JSON.stringify(args) + JSON.stringify(whileTrue) + JSON.stringify(guid))
+          .digest('hex');
+
+        try {
+          const latestSnapshot = await cache.get<T>(id);
+
+          logger.trace('cacheSnapshot', latestSnapshot);
+          const results = store.read(query, latestSnapshot.flatMap((snapshot) => Option.of(snapshot.time)), ...args);
+
+          const aggregatedAt = new Date();
+
+          const aggregatedResult = await reduceWhile<Event<EventData, EventContext<any>>, Option<T>>(
```

Comment:
I wanted to use R.reduceWhile here, but it doesn't seem to work with AsyncIterator..